### PR TITLE
[BIT] 引擎补齐：zeropad2d, softmax, quant.weight_only_linear, __sub__

### PR DESCRIPTION
**Pull request by Cutelemon6**
### paddle.nn.functional.zeropad2d
要求参数 padding 不能为负数，且不能太大，限制在了0 ~ 10

### paddle.nn.functional.softmax
有一个配置是 bfloat16，在反向时会产生一个 bf16 的 output 和 float32 的 output_grad，这样在 paddle.grad 中会选择 float32 的内核，属于内核错误，放入 fresh report；另外有参数 axis，需要保证下标在 [-len(shape x), len(shape x)) 之间，这个已修复。

### paddle.nn.quant.weight_only_linear
只能在 sm80 或者 sm86 的 gpu 上运行，放入 fresh report

### paddle.Tensor.\_\_sub__
0-size Tensor 配置下，执行 `try:paddle.base.core.eager._for_test_check_cuda_error()` 后抛出错误，放入 fresh report

### CODE REVIEW

File: report/fresh_report/test_log_paddle.nn.quant.weight_only_linear.log [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+[paddle error] paddle.nn.quant.weight_only_linear(Tensor([1, 1, 64],"float16"), Tensor([128, 64],"int8"), bias=Tensor([256],"float16"), weight_scale=Tensor([256],"float16"), weight_dtype="int4", group_size=-1, ) 
+ (InvalidArgument) Currently weightonly linear grad only support arch = 80 or 86. 
+  [Hint: Expected ((arch == 80) || (arch == 86)) == true, but received ((arch == 80) || (arch == 86)):0 != true:1.] (at ../paddle/phi/infermeta/backward.cc:1765)
+
+paddle.nn.quant.weight_only_linear(Tensor([1, 1, 64],"float16"), Tensor([256, 64],"int8"), bias=Tensor([256],"float16"), weight_scale=Tensor([256],"float16"), weight_dtype="int8", group_size=-1, )
+
+[paddle error] paddle.nn.quant.weight_only_linear(Tensor([1, 1, 64],"float16"), Tensor([256, 64],"int8"), bias=Tensor([256],"float16"), weight_scale=Tensor([256],"float16"), weight_dtype="int8", group_size=-1, ) 
+ (InvalidArgument) Currently weightonly linear grad only support arch = 80 or 86. 
+  [Hint: Expected ((arch == 80) || (arch == 86)) == true, but received ((arch == 80) || (arch == 86)):0 != true:1.] (at ../paddle/phi/infermeta/backward.cc:1765)
```
Comment (wanghuancoder) [inline report/fresh_report/test_log_paddle.nn.quant.weight_only_linear.log:11]: 这个看起来是你的机器环境不行，他要求V100、A100~

Comment (wanghuancoder) [reply report/fresh_report/test_log_paddle.nn.quant.weight_only_linear.log:11]: 可以先放在这儿，回头再看

**Review by wanghuancoder — APPROVED** [review]
LGTM
